.github/workflows/gmc-e2e.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+name: gmc e2e test
+
+on:
+  pull_request:
+    branches: [master]
+    types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
+    paths:
+      - microservices-connector/**
+      - .github/workflows/gmc-e2e.yaml
+  workflow_dispatch:
+
+# If there is a new commit, the previous jobs will be canceled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  kind:
+    runs-on: k8s
+    steps:
+      - name: Checkout out Repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Run e2e test for microservices-connector
+        run: |
+          .github/workflows/scripts/e2e/gmc_test.sh validate_chatqna
+      - name: Clean up e2e test for microservices-connector
+        run: |
```

Comment:
Add a `if: always()` before `run:`. We want the clean up happens every time whether the previous test fail and success. Refer to: https://github.com/opea-project/GenAIInfra/blob/main/.github/workflows/chart-e2e.yaml#L107